lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -836,6 +837,27 @@ joinStakePool ctx spl (ApiT pid) (ApiT wid) (ApiWalletPassphrase (ApiT pwd)) = d
   where
     liftE = throwE . ErrJoinStakePoolNoSuchWallet
 
+joinStakePoolFee
+    :: forall ctx s t n k.
+        ( DelegationAddress n k
+        , Buildable (ErrValidateSelection t)
+        , s ~ SeqState n k
+        , k ~ ShelleyKey
+        , HardDerivation k
+        , ctx ~ ApiLayer s t k
+        )
+    => ctx
+    -> StakePoolLayer IO
+    -> ApiT PoolId
+    -> ApiT WalletId
+    -> Handler ApiFee
+joinStakePoolFee ctx _spl (ApiT _pid) (ApiT wid) = do
```

Comment:
At first glance, it seems as though the following arguments can be completely omitted:

1. `_spl :: StakePoolLayer IO`
2. `_pid :: PoolId`

The first argument can be omitted by changing the definition of `stakePools` to:
```hs
stakePools ctx spl =
    listPools spl
    :<|> joinStakePool ctx spl
    :<|> joinStakePoolFee ctx
    :<|> quitStakePool ctx
```

The second argument leads me to another question:

Will the fee calculation always be independent of the pool ID? If the answer is yes, then is there any need for the endpoint to include the pool ID?

Reply:
I've indeed removed the pool id from the endpoint path and from the handler arguments.